robot_sort/robot_sort.py

```diff
@@ -96,8 +96,18 @@ def sort(self):
         """
         Sort the robot's list.
         """
-        # Fill this out
-        pass
+        for step in range(len(self._list)):
+            key = self._list[step]
+
+            j = step - 1
+
+            #comparing the key with each element on the left until a smaller element is found
+            while j >= 0 and key < self._list[j]:
+                self._list[j + 1] = self._list[j]
+                j = j - 1
+
+                #placing the key after the element
+                self._list[j + 1] = key
```

Comment:
You cannot modify the robot's properties. Anything with an underscore like `self._list`  or even `self._item` cannot be used to solve this challenge.